dashboard/app/reporting_email.go

```diff
@@ -76,6 +76,13 @@ type EmailConfig struct {
 	MailMaintainers    bool
 	DefaultMaintainers []string
 	SubjectPrefix      string
+	// DiscussionSources is the ordered list of (Cc'd email, DiscussionSource) pairs.
+	DiscussionSources []EmailConfigDiscussion
+}
+
+type EmailConfigDiscussion struct {
+	Email  string
+	Source dashapi.DiscussionSource
```

Comment:
If we subscribe several dozens of lists, we will need to keep all of them here as well. And we will need to update the config each time we subscribe a new list. Is it possible to somehow avoid this and infer things automatically?

Reply:
By taking Cc/To and comparing it with the known mailing lists we can be sure that the email also exists there.

Here are, for example, a few random messages from different mailing lists on lore:
https://lore.kernel.org/all/20230324181741.13908-1-michael.christie@oracle.com/raw (dm-devel)
https://lore.kernel.org/all/20230316225926.494921-1-rmoar@google.com/raw (kernelci)

I don't really see anything that could reliably connect them and indicate that we can safely construct a `https://lore.kernel.org/all/%v/T` link.

At best we could look in `To` not for individual emails, but for patterns e.g. `.*@vger.kernel.org`, but I'm not sure that all addresses at `vger.kernel.org` are mailing lists.